crates/db_views/src/combined/modlog_combined_view.rs

```diff
@@ -162,6 +161,8 @@ impl ModlogCombinedQuery {
 
     let mod_person = self.mod_person_id.unwrap_or(PersonId(-1));
     let show_mod_names = !(self.hide_modlog_names.unwrap_or_default());
+
+    // Unfortunately, you can't use auto_type or abstract the join, because of this as_sql
```

Comment:
You can get auto_type working here by adding an explicit type to the variable below.

Reply:
I tried messing with it for a long time, but it uses `Bound<sql_type::Bool, bool>`, which isn't public.

Reply:
You want to use [`AsExprOf`](https://docs.diesel.rs/2.2.x/diesel/dsl/type.AsExprOf.html) as shown as return type [by the documentation](https://docs.diesel.rs/2.2.x/diesel/expression/trait.IntoSql.html#method.as_sql).

Reply:
I couldn't figure it out, the trait bounds weren't satisfied for some reason.

```rust
fn bool_expr() -> AsExprOf<diesel::sql_types::Bool, bool> {
  true.as_sql::<diesel::sql_types::Bool>()
}

#[diesel::dsl::auto_type(no_type_alias)]
fn joins() -> _ {
  let test = bool_expr();
  test.or(....
  
}
...
```
Errors

> the trait `diesel::Expression` is not implemented for `diesel::sql_types::Bool`, which is required by `diesel::sql_types::Bool: diesel::expression::AsExpression<bool>

Reply:
The linked function signature shows that `as_sql` takes a **reference** to self as argument and the return type also takes a reference as generic argument. There is the `into_sql` as owned variant. 

That written: You don't need a separate function, just aus a type ascription.

Reply:
Thx, the `.into_sql::<diesel::sql_types::Bool>` did it.